renku_notebooks/api/classes/s3mount.py

```diff
@@ -126,14 +126,26 @@ def get_manifest_patches(
     @property
     def bucket_exists(self):
         try:
-            self.client.head_bucket(Bucket=self.bucket)
+            res = self.client.head_bucket(Bucket=self.bucket)
         except (ClientError, EndpointConnectionError, NoCredentialsError, ValueError):
             current_app.logger.warning(
                 f"Failed to confirm bucket {self.bucket} for endpoint {self.endpoint} exists"
             )
             return False
-        else:
-            return True
+        # INFO: If the region is not the default (us-east-1) and it is not speicifed explicitly in
+        # the endpoint then datashim has trouble mounting the bucket even though boto can find it.
+        amz_bucket_region = (
+            res.get("ResponseMetadata", {})
+            .get("HTTPHeaders", {})
+            .get("x-amz-bucket-region")
+        )
+        if (
+            amz_bucket_region
+            and amz_bucket_region != "us-east-1"
+            and amz_bucket_region not in self.endpoint
```

Comment:
why do we fail here? If we can determine the true location of the bucket, shouldn't we just go ahead (like boto does)? I guess that would mean amending the endpoint with the actual zone.

Reply:
To me it seemed a bit risky and unexpected (from the users perspective) to change a request parameter that came directly from the user. But yes we can do this if we hardcode a dictionary of regions mapped to the correct endpoints.

The problem is that if the urls for these endpoints change then things will break and we will have to update the code.

Reply:
Ok I will change the endpoint in this case then